### PR TITLE
doc changes (small)

### PR DESCRIPTION
**Pull request by ryanbas21**
grammar

Just browsing through docs, made small grammar fixes when I saw them. Looking at it as if I was a complete beginner, I was curious if the term higher order component should be assumed already, so I added that small definition in parentheses. 

Thanks for contributing to react-virtualized!

Here is a short checklist of additional things to keep in mind before submitting:
* Please make sure your pull request description makes it very clear what you're trying to accomplish. If it's a bug fix, please also provide a failing test case (if possible). In either case, please add additional unit test coverage for your changes. :)
* Run tests locally (`npm test`) to ensure that your change did not break linting or functionality.
* Be sure you have notifications setup so that you'll see my code review responses. (I may ask you to make some adjustments before merging.)

### CODE REVIEW

File: docs/ArrowKeyStepper.md [hunk 1]
```diff
@@ -1,7 +1,7 @@
 ArrowKeyStepper
 ---------------
 
-High-order component that decorates another virtualized component and responds to arrow-key events by scrolling one row or column at a time.
+A High-order component (A component that takes a component and returns a component) that decorates another virtualized component and responds to arrow-key events by scrolling one row or column at a time.
```
Comment (TrySound) [inline docs/ArrowKeyStepper.md]: I'd say it's not higher order component. Can you call it `A component with children as function that wraps...`?

File: docs/ScrollSync.md [hunk 2]
```diff
@@ -1,7 +1,7 @@
 ScrollSync
 ---------------
 
-High order component that simplifies the process of synchronizing scrolling between two or more virtualized components.
+High-order component that simplifies the process of synchronizing scrolling between two or more virtualized components.
```
Comment (TrySound) [inline docs/ScrollSync.md]: `A component with children as function that...`

**Comment by ryanbas21** [comment]
@TrySound So I've actually never heard the term `high-order` component. What would the definition of it be? I even asked at work and no one knew.

**Comment by TrySound** [comment]
Higher order component is function which wraps react component and returns new one
```js
import { pure } from 'recompose'

const enhanced = pure(
  ({ width, height }) => <div style={{ width, height }}></div>
)
```

it's mostly not quite useful since introduces props collision. Render prop or children as function is more clean way to do things this time. Also children as functions is used for react stuff (context api).

**Comment by ryanbas21** [comment]
I'm confused now, is `high order === higher order`? Or are they different terms. I understand a `higher order component` as a component that takes a `Component` and returns a `Component`, usually applying some new props to said component. 

So when we say a `high-order` component are we saying more like a render prop?

**Comment by TrySound** [comment]
Probably spelling

File: docs/ArrowKeyStepper.md (hunk 1 above)
Comment (bvaughn) [reply docs/ArrowKeyStepper.md]: Yeah, I guess these docs are a little out of date. `ArrowKeyStepper` used to be implemented more as a HOC but changed to use the function children / render props a long time ago. Maybe the docs could be updated to say something more like...

> Decorates another react-virtualized component such as `Grid` or `List`. Passes `scrollToColumn` and `scrollToRow` props to a render function to enable user to navigate content using arrow keys.

File: docs/ScrollSync.md (hunk 2 above)
Comment (bvaughn) [reply docs/ScrollSync.md]: Maybe something like

> Decorates multiple virtualized components such as `Grid` or `List` in order to synchronize their scroll positions. Passes synchronized scroll offsets to a render prop.

**Comment by chris-fa** [comment]
@ryanbas21 since you're doing a PR on docs, what do you think on documenting that the `<Table>` `rowGetter` property should not receive a null nor undefined, but a blank object, too?

Here's the issue for reference: https://github.com/bvaughn/react-virtualized/issues/1012, check my last comment

**Comment by ryanbas21** [comment]
@chris-fa sure! I'll check it out later tonight.

File: docs/ScrollSync.md (hunk 2 above)
Comment (ryanbas21) [reply docs/ScrollSync.md]: @bvaughn Cool, this makes sense. is this something that I should include in each instance of `high-order` or as a separate glossary type page?

**Comment by aem** [comment]
As a project it makes sense to settle on a terminology, and "higher-order" is the accepted term afaik. While "high-order" is an absolute term, "higher-order" references the fact that it's simply a function of a higher order than the function it wraps, which makes sense to me.

> since you're doing a PR on docs, what do you think on documenting that the <Table> rowGetter property should not receive a null nor undefined, but a blank object, too?

@chris-fa we should keep changes focused, it doesn't make sense to lump more docs changes into this PR just because it's open. Unless those changes are related to these, let's open a separate PR for those

**Comment by ryanbas21** [comment]
@aem I generally think higher order in terms of components being it will take and return although I guess you could say it has to take and or return a component. It definitely would be a more understood term, but when I first read high-order I and others I asked just assumed it was another way of saying higher order, but with how this discussion has gone I haven't felt that its the same definition.

**Comment by ryanbas21** [comment]
FWIW - it seems the markdown plugin I installed also reformatted the markdown to make stuff line up - if that is unwanted I can revert this.

**Comment by ryanbas21** [comment]
closing cause i blew up my git.

**Pull request closed without merging**